api/http/api_useradm.go

```diff
@@ -404,3 +406,29 @@ func (u *UserAdmApiHandlers) DeleteTokensHandler(w rest.ResponseWriter, r *rest.
 		rest_utils.RestErrWithLogInternal(w, r, l, err)
 	}
 }
+
+func (u *UserAdmApiHandlers) SaveSettingsHandler(w rest.ResponseWriter, r *rest.Request) {
+	ctx := r.Context()
+
+	l := log.FromContext(ctx)
+
+	var settings map[string]interface{}
+
+	err := r.DecodeJsonPayload(&settings)
+	if err != nil {
+		rest_utils.RestErrWithLog(w, r, l, errors.Wrap(err, "failed to parse request body"), http.StatusBadRequest)
```

Comment:
do we want to return error from json decoder? or something more generic like setting object must be json or something like that?

Reply:
yeah, it's enough to say 'cannot parse as json', fixed